web/src/modules/topic/components/EditableNode/EditableNode.tsx

```diff
@@ -26,6 +27,13 @@ export const EditableNode = ({ node }: { node: Node }) => {
   const color = theme.palette[node.type].main;
   const NodeIcon = nodeDecoration.NodeIcon;
 
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
```

Comment:
Yeah your screenshot is what I mean. Your code is fine as is for meeting React's standards for being top level of the function (not within any blocks), but I personally prefer _additionally_ grouping hooks together in the first few lines of the function, so that it's immediately obvious where we shouldn't be using blocks.